warehouse/migrations/versions/08ccc59d9857_add_manual_password_reset_prohibition.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+add manual password reset prohibition
+
+Revision ID: 08ccc59d9857
+Revises: 10825786b3df
+Create Date: 2021-07-13 14:40:16.604041
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+
+revision = "08ccc59d9857"
+down_revision = "10825786b3df"
+
+
+def upgrade():
+    op.add_column(
+        "users",
+        sa.Column(
+            "prohibit_password_reset",
+            sa.Boolean(),
+            server_default=sa.text("false"),
+            nullable=False,
+        ),
+    )
```

Comment:
I'm not an alambic expert, but I believe adding a non-nullable column will likely force PG to rewrite every row (522,487 according to the front page). I trust your judgement & experience on this, but on my part, I'd tend to say it might be worth it to try and run this migration during a time where a few seconds freeze of the DB will likely not be seen.

(the way of doing this properly is most likely overkill if we don't have several million rows)
(and again, maybe alembic actually takes care of this, I don't know)

Reply:
we've run similar migrations using the `server_default` incantation in the past with no impact, see `is_psf_staff` addition a few weeks back. it's more on the PG side than the alembic/sqlachemy side, but using a server_default Postgres knows what to do.